wrapper/src/libvmaf.h

```diff
@@ -32,92 +34,94 @@
+    VMAF_PIX_FMT_YUV444P10LE                           = 5,
+    VMAF_PIX_FMT_UNKNOWN                               = 6,
 };
 
-class StatVector
+typedef struct VmafPicture
 {
-public:
-    StatVector();
-    StatVector(std::vector<double> l);
-    std::vector<double> getVector();
-    double mean();
-    double minimum();
-    double harmonic_mean();
-    double second_moment();
-    double percentile(double perc);
-    double var();
-    double std();
-    void append(double e);
-    double at(size_t idx);
-    size_t size();
-private:
-    std::vector<double> l;
-    void _assert_size();
+    float *data[3];
+    unsigned int w[3], h[3];
+    unsigned int stride_byte[3];
+    enum VmafPixelFormat pix_fmt;
+} VmafPicture;
+
+enum VmafModelSetting {
+    VMAF_MODEL_SETTING_NONE                            = (1 << 0),
+    VMAF_MODEL_SETTING_ENABLE_TRANSFORM                = (1 << 1),
+    VMAF_MODEL_SETTING_DISABLE_CLIP                    = (1 << 2),
+    VMAF_MODEL_SETTING_ENABLE_CONF_INTERVAL            = (1 << 3),
 };
 
+typedef struct {
+    char *name;
+    char *path;
+    int vmaf_model_setting;
+} VmafModel;
 
-class Result
-{
-public:
-    Result();
-    void set_scores(const std::string &key, const StatVector &scores);
-    StatVector get_scores(const std::string &key);
-    bool has_scores(const std::string &key);
-    double get_score(const std::string &key);
-    std::vector<std::string> get_keys();
-    void setScoreAggregateMethod(ScoreAggregateMethod scoreAggregateMethod);
-private:
-    std::map<std::string, StatVector> d;
-    ScoreAggregateMethod score_aggregate_method;
-};
+typedef struct {
 
-class IVmafQualityRunner {
-public:
-    virtual Result run(Asset asset, int(*read_frame)(float *ref_data, float *main_data, float *temp_data,
-        int stride, void *user_data), void *user_data, bool disable_clip, bool enable_transform,
-        bool do_psnr, bool do_ssim, bool do_ms_ssim, int n_thread, int n_subsample) = 0;
-    virtual ~IVmafQualityRunner() {}
-};
+    unsigned int width;
+    unsigned int height;
+    unsigned int num_models;
+    unsigned int default_model_ind; // useful to tell between the default model and additional models (if any)
 
-class VmafQualityRunnerFactory {
-public:
-    static std::unique_ptr<IVmafQualityRunner> 
-        createVmafQualityRunner(const char *model_path, bool enable_conf_interval);
-};
+    char *log_path;
+    int vmaf_feature_mode_setting;
+
+    VmafModel vmaf_model[MAX_NUM_VMAF_MODELS];
+    VmafFeatureCalculationSetting vmaf_feature_calculation_setting;
+
+    enum VmafPixelFormat pix_fmt;
+    enum VmafLogFmt log_fmt;
+    enum VmafPoolingMethod pool_method;
 
+} VmafSettings;
+
+int compute_vmaf(double* vmaf_score,
+                 int (*read_frame)(float *ref_data, float *main_data, float *temp_data, int stride_byte, void *user_data),
+                 int (*read_vmaf_picture)(VmafPicture *ref_vmaf_pict, VmafPicture *dis_vmaf_pict, float *temp_data, void *user_data),
```

Comment:
Can you write doxygen style API docs for the library functions/types in the header file? I have questions about how to use `compute_vmaf()` for example: 

- Why are there two `read_*` function pointers?
- Why is there only one `vmaf_score` returned, when multiple models are used? What about the confidence interval, etc? Is that available in data somewhere?

Reply:
Sure, I will be adding some documentation. Regarding the two questions:
1. there are two function pointers just for us to look at the before vs. after state. Will be removing this from the signature soon.
2. So far, the convention has been to return only a single value (vmaf_score) to the caller and write to a log file everything else. We could extend that to return something like a pointer to a result struct. Thinking of ways to do that.

Reply:
Created a struct VmafOutput which can be extended in the future as we see fit. For now, compute_vmaf is changed to return a pointer to that struct.

Reply:
> Created a struct VmafOutput which can be extended in the future as we see fit. For now, compute_vmaf is changed to return a pointer to that struct.

What about the scores from the additional models?

Reply:
The scores from additional models, along with other data that we want to return can be part of that struct.